test/unitary/xtypes/struct_type.cpp

```diff
@@ -0,0 +1,419 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <xtypes/xtypes.hpp>
+#include <iostream>
+
+#include <cmath>
+#include <bitset>
+
+using namespace std;
+using namespace eprosima::xtypes;
+
+static const uint8_t UINT8          = 250;
+static const int16_t INT16          = -32760;
+static const uint16_t UINT16        = 65530;
+static const int32_t INT32          = -2147483640;
+static const uint32_t UINT32        = 4294967290;
+static const int64_t INT64          = -9223372036854775800;
+static const uint64_t UINT64        = 18446744073709551610ULL;
+static const float FLOAT            = 3.1415927410125732421875f;
+static const double DOUBLE          = 3.1415926535897931159979631875;
+static const long double LDOUBLE    = 3.14159265358979321159979631875l;
+static const char CHAR              = 'f';
+static const wchar_t WCHAR          = 34590;
+
+static const std::string INNER_STRING_VALUE = "lay_down_and_cry";
+static const std::string INNER_SEQUENCE_STRING = "another_prick_in_the_wall";
+static const std::string SECOND_INNER_STRING = "paint_it_black";
+
+static const float STRUCTS_SIZE = 1E1;
+
+/**********************************
+ *        StructType Tests        *
+ **********************************/
+
+TEST (StructType, primitive_struct)
+{
+    StructType st("struct_name");
```

Comment:
This is pending to refactorize from Francesco, right?

Reply:
Added some helper functions to ease code reading.

Reply:
Great!